ts/components/settings/section/CategoryConversations.tsx

```diff
@@ -81,13 +83,50 @@ const AudioMessageAutoPlaySetting = () => {
   );
 };
 
+const EnterKeyFunctionSetting = () => {
+  const forceUpdate = useUpdate();
+
+  const initialSetting = window.getSettingValue(SettingsKey.hasShiftSendEnabled) || false;
+
+  const items = [
+    {
+      label: window.i18n('enterSendNewMessageDescription'),
+      value: false,
+    },
+    {
+      label: window.i18n('enterNewLineDescription'),
+      value: true,
+    },
+  ];
+
+  return (
+    <SessionSettingsItemWrapper
+      title={window.i18n('enterKeySettingTitle')}
+      description={window.i18n('enterKeySettingDescription')}
+      inline={false}
+    >
+      <SessionRadioGroup
+        initialItem={initialSetting}
+        group={SettingsKey.hasShiftSendEnabled} // make sure to define this key in your SettingsKey enum
+        items={items}
+        /* eslint-disable @typescript-eslint/no-misused-promises */
+        onClick={async (selectedRadioValue: string | boolean) => {
+          await window.setSettingValue(SettingsKey.hasShiftSendEnabled, selectedRadioValue);
+          forceUpdate();
+        }}
```

Comment:
once you got rid of the forceUpdate(), and because that onClick will only be a oneliner of a promise,
you could
- remove the `eslint-disable @typescript-eslint/no-misused-promises`
- remove the async from the onClick itself
- make the window.setSettingValue not `awaited`, but `void`